scapy/layers/llmnr.py

```diff
@@ -57,15 +59,19 @@ def hashret(self):
         return struct.pack("!H", self.id)
 
     def mysummary(self):
-        if self.an:
-            return "LLMNRResponse '%s' is at '%s'" % (
-                self.an[0].rrname.decode(errors="backslashreplace"),
-                self.an[0].rdata,
-            ), [UDP]
-        if self.qd:
-            return "LLMNRQuery who has '%s'" % (
-                self.qd[0].qname.decode(errors="backslashreplace"),
-            ), [UDP]
+        s = self.__class__.__name__
+        if self.qr:
+            if self.an and isinstance(self.an[0], DNSRR):
+                s += " '%s' is at '%s'" % (
+                    self.an[0].rrname.decode(errors="backslashreplace"),
+                    self.an[0].rdata,
+                )
+        else:
+            if self.qd and isinstance(self.qd[0], DNSQR):
+                s += " who has '%s'" % (
+                    self.qd[0].qname.decode(errors="backslashreplace"),
+                )
```

Comment:
```suggestion
        if self.qr:
            if self.an and isinstance(self.an[0], DNSRR):
                s += " '%s' is at '%s'" % (
                    self.an[0].rrname.decode(errors="backslashreplace"),
                    self.an[0].rdata,
                )
            else:
                s += " malformed answer"
        elif self.qd and isinstance(self.qd[0], DNSQR):
            s += " who has '%s'" % (
                self.qd[0].qname.decode(errors="backslashreplace"),
            )
        else:
            s += " malformed query"
```

Reply:
Makes sense. Would it be OK if kept the class name and appended `[malformed]`? It would be consistent with `tshark` and look like
```sh
LLMNRQuery [malformed]
LLMNRResponse [malformed]
```
(I think it would probably be easier to grep those too)